safe/gui/tools/print_report_dialog.py

```diff
@@ -0,0 +1,486 @@
+        """Helper method for populating template combobox.
+
+        :param unwanted_templates: List of templates that isn't an option.
+        :type unwanted_templates: list
+
+        .. versionadded: 4.3.0
+        """
+        templates_dir = QtCore.QDir(path)
+        templates_dir.setFilter(
+            QtCore.QDir.Files |
+            QtCore.QDir.NoSymLinks |
+            QtCore.QDir.NoDotAndDotDot)
+        templates_dir.setNameFilters(['*.qpt', '*.QPT'])
+        report_files = templates_dir.entryList()
+        for unwanted_template in unwanted_templates:
+            if unwanted_template in report_files:
+                report_files.remove(unwanted_template)
+
+        for f in report_files:
+            self.template_combo.addItem(
+                QtCore.QFileInfo(f).baseName(), path + '/' + f)
+
+    def restore_state(self):
+        """Reinstate the options based on the user's stored session info.
+
+        .. versionadded: 4.3.0
+        """
+        settings = QtCore.QSettings()
+
+        flag = bool(settings.value(
+            'inasafe/useDefaultTemplates', True, type=bool))
+        self.default_template_radio.setChecked(flag)
+        self.custom_template_radio.setChecked(not flag)
+
+        try:
+            default_template_path = resources_path(
+                'qgis-composer-templates', 'inasafe-map-report-portrait.qpt')
+            path = settings.value(
+                'inasafe/lastTemplate',
+                default_template_path,
+                type=str)
+            self.template_combo.setCurrentIndex(
+                self.template_combo.findData(path))
+        except TypeError:
+            self.template_combo.setCurrentIndex(2)
+
+        try:
+            path = settings.value('inasafe/lastCustomTemplate', '', type=str)
+        except TypeError:
+            path = ''
+        self.template_path.setText(path)
+
+    def save_state(self):
+        """Store the options into the user's stored session info.
+
+        .. versionadded: 4.3.0
+        """
+        settings = QtCore.QSettings()
+        settings.setValue(
+            'inasafe/useDefaultTemplates',
+            self.default_template_radio.isChecked())
+        settings.setValue(
+            'inasafe/lastTemplate',
+            self.template_combo.itemData(self.template_combo.currentIndex()))
+        settings.setValue(
+            'inasafe/lastCustomTemplate', self.template_path.text())
+
+    def retrieve_paths(self, products, report_path, suffix=None):
+        """Helper method to retrieve path from particular report metadata.
+
+        :param products: Report products.
+        :type products: list
+
+        :param report_path: Path of the IF output.
+        :type report_path: str
+
+        :param suffix: Expected output product file type (extension).
+        :type suffix: str
+
+        :return: List of absolute path of the output product.
+        :rtype: list
+        """
+        paths = []
+        for c in products:
```

Comment:
Avoid single character as variable if possible. It's better to use `for product in products`. And `c` is not obvious, not like `for p in paths`